### PR TITLE
Fix coverity issues

### PR DESCRIPTION
**Pull request by banasraf**
Signed-off-by: Rafal <rbanas@nvidia.com>

## Category:

Bug fix

## Description:

It fixes issues found by coverity

### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/imgcodec/util/convert.h [hunk 1]
```diff
@@ -274,11 +274,6 @@ void Convert(Out *out, const int64_t *out_strides, int out_channel_dim, DALIImag
     out_channel_stride = -out_channel_stride;
   }
 
-  if (out_format == DALI_ANY_DATA) {
-    // Conversion to DALI_ANY_DATA is a no-op
-    out_format = in_format;
-  }
```
Comment (mzient) [inline dali/imgcodec/util/convert.h:280]: Why so?

Comment (banasraf) [reply dali/imgcodec/util/convert.h:280]: It repeats condition from L254

File: dali/imgcodec/util/convert.cc [hunk 2]
```diff
@@ -26,6 +26,10 @@ namespace imgcodec {
 void Convert(SampleView<CPUBackend> out, TensorLayout out_layout, DALIImageType out_format,
              ConstSampleView<CPUBackend> in, TensorLayout in_layout, DALIImageType in_format,
              ROI roi, Orientation orientation) {
+  DALI_ENFORCE(in_layout.find('H') >= 0 && in_layout.find('W') >= 0,
+               "Input layout has to contain at least H and W dimensions.");
```
Comment (mzient) [inline dali/imgcodec/util/convert.cc:30]: Technically, there's no such requirement for the input layout.

File: dali/imgcodec/util/convert.cc [hunk 3]
```diff
@@ -26,6 +26,10 @@ namespace imgcodec {
...
+  DALI_ENFORCE(out_layout.find('H') >= 0 && out_layout.find('W') >= 0,
+               "Output layout has to contain at least H and W dimensions.");
```
Comment (mzient) [inline dali/imgcodec/util/convert.cc:32]: I guess at this point it should be:
```suggestion
  int h_dim = ImageLayoutInfo::DimIndex(out_layout, 'H');
  int w_dim = ImageLayoutInfo::DimIndex(out_layout, 'W');
  DALI_ENFORCE(h_dim >= 0 && w_dim >= 0,
               "Output layout has to contain at least H and W dimensions.");
```
...and you could remove the
```
      auto h_dim = ImageLayoutInfo::DimIndex(out_layout, 'H');
      auto w_dim = ImageLayoutInfo::DimIndex(out_layout, 'W');
```
from the TYPE_SWITCH below.

File: dali/imgcodec/util/convert.h (hunk 1 above)
Comment (mzient) [reply dali/imgcodec/util/convert.h:280]: OK

**Review by mzient — APPROVED** [review]
LGTM, with some minor remarks

**Review by szalpal — APPROVED** [review]

**Pull request merged**